### ISSUE TITLE
When using the colon notation in Quarto, such as when using tab sets, the completion and indentation of R chunks stop functioning properly.

### ISSUE THREAD

**Issue by ujtwr** [rstudio/rstudio#14739]
### System details

    RStudio Edition : Server
    RStudio Version : 2024.04.1+748
    OS Version        :  Ubuntu (server) + Win11 Chorome (client)
    R Version          :  4.4.0
    Quarto             : 1.4.551

### Steps to reproduce the problem

1. Create new Quarto file (.qmd)
2. Insert something elements using the colon notation like bellow:

```
::: {.panel-tabset}

:::
```

3. Insert new R chunk after this
4. In this new chunk, tab completion and indentation when pressing Enter do not work correctly. Also, re-indentation using Ctrl+i does not function.

### Describe the problem in detail

The correct indentation is as follows:

````
```{r}
iris %>% 
  as_tibble() %>% 
  filter(
    Species == "setosa"
  ) %>% 
  count()
```
````

Because the indentation is not set correctly, it looks like this. Additionally, tab completion for variables does not work.

Also, when executing the code with Ctrl+Enter while the cursor is in the middle of the code, depending on the cursor's position, the code connected by pipes may not be recognized, resulting in an error.

````
```{r}
iris %>% 
as_tibble() %>% 
filter(
Species == "setosa"
) %>% 
count()

```
````

![image](https://github.com/rstudio/rstudio/assets/42717412/97eaaaf7-9802-4482-acbd-38fd41780065)

### Describe the behavior you expected

Completion and indentation will function correctly.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
Regression caused by https://github.com/rstudio/rstudio/pull/14558.

See also other similar issues:

- [Reindent & Reformat Lines stopped working in Quarto Callout Blocks #14640](https://github.com/rstudio/rstudio/issues/14640)
- [Function after an empty div breaks source code blocks in Quarto document #14652](https://github.com/rstudio/rstudio/issues/14652)

**Comment by gtritchie** [comment]
This was already fixed in 2024.04.2 (Chocolate Cosmos patch). Presumably fixed by https://github.com/rstudio/rstudio/pull/14744.

We should do a quick double-check of this against Cranberry Hibiscus.

**Comment by ronblum** [comment]
Verified in RStudio Desktop 2024.07.0-daily+303 and 2024.04.2+764 on Windows 11. Checked both the reindentation and completions.
